From f24f9cbb317e3fe1806a426d4a0e08cf4775cb4d Mon Sep 17 00:00:00 2001
From: Li Jun <B47624@freescale.com>
Date: Mon, 29 Sep 2014 10:22:52 +0800
Subject: [PATCH 0674/1221] MLK-9617-11 usb: common: otg: use a_set_b_hnp_en
 instead of b_hnp_enable

This patch replaces b_hnp_enable with a_set_b_hnp_en to comply with state diagram
of USB and EH spec.

Signed-off-by: Li Jun <b47624@freescale.com>
(cherry picked from commit 4734ed1f813959a7544ad744d60ab119efb04ba9)
[zou:Original patch taken from
git.freescale.com/imx/fsl-arm-yocto-bsp.git-b imx-3.14.52-1.1.0_ga]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 drivers/usb/common/usb-otg-fsm.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/common/usb-otg-fsm.c b/drivers/usb/common/usb-otg-fsm.c
index cd3cf83..31a8777 100644
--- a/drivers/usb/common/usb-otg-fsm.c
+++ b/drivers/usb/common/usb-otg-fsm.c
@@ -342,9 +342,9 @@ int otg_statemachine(struct otg_fsm *fsm)
 			otg_set_state(fsm, OTG_STATE_A_VBUS_ERR);
 		break;
 	case OTG_STATE_A_SUSPEND:
-		if (!fsm->b_conn && fsm->otg->host->b_hnp_enable)
+		if (!fsm->b_conn && fsm->a_set_b_hnp_en)
 			otg_set_state(fsm, OTG_STATE_A_PERIPHERAL);
-		else if (!fsm->b_conn && !fsm->otg->host->b_hnp_enable)
+		else if (!fsm->b_conn && !fsm->a_set_b_hnp_en)
 			otg_set_state(fsm, OTG_STATE_A_WAIT_BCON);
 		else if (fsm->a_bus_req || fsm->b_bus_resume)
 			otg_set_state(fsm, OTG_STATE_A_HOST);
-- 
1.7.5.4

